core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -68,13 +82,26 @@
     private static final Logger LOG = LoggerFactory.getLogger(Factory.class);
     private static final AtomicBoolean CONF_LOGGED = new AtomicBoolean(false);
 
+    protected static final FileSystem.Cache FILESYSTEM_CACHE = new FileSystem.Cache();
+
     private Factory() {} // prevent instantiation
 
-    public static FileSystem get(AlluxioConfiguration alluxioConf) {
-      return get(FileSystemContext.create(alluxioConf));
+    public static FileSystem get() {
+      return get(new Subject()); // Use empty subject
+    }
+
+    public static FileSystem get(Subject subject) {
+      Preconditions.checkNotNull(subject, "subject");
+      AlluxioConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
+      FileSystemKey key = new FileSystemKey(subject, conf);
+      return FILESYSTEM_CACHE.get(key);
     }
 
-    public static FileSystem get(FileSystemContext context) {
+    public static FileSystem create(AlluxioConfiguration alluxioConf) {
```

Comment:
We should comment that the `create` methods don't cache, but the `get` ones do.

Reply:
Will update